### PR TITLE
Deal with CommandBar app labels

### LINKED ISSUE #3391: With the new commandbar, I don't get tooltips on any external controls
### Steps to reproduce the bug

1. Add any external tools, and pin to the bar.
2. See that they don't get tooltips.

### PR DESCRIPTION
**Pull request by timkur**
## Summary of the pull request

We were using the standard behavior for the CommandBar to show text labels when the bar was expanded.

image

It kind of looks bad, especially for tool names that take up multiple lines. It also requires the bar to be enlarged to show the text labels, and if the bar isn't enlarged enough, the popup menu goes up (and off the screen) instead of down.

Additionally, using the dynamic overflow feature of the command bar caused the list of tools to be fragmented... a list of pinned tools and a list of non-pinned tools.

This updates the CommandBar to no longer try and put labels on the expanded CommandBar, and turns off dynamic overflow, and will only show one list of External Tools (SecondaryCommands for those looking at the code). The order of these tools should stay consistent, and all tools will show up in the dropdown menu regardless if the tool is pinned or not.

This also adds tooltips to the bar.

image

## Detailed description of the pull request / Additional comments

This also removes the MenuIcon property from an ExternalTool. This exposed ImageIcon is good for "1 use only" and is too tempting to just use (and unfortunately there isn't an ImageIcon.Copy() or Clone().

This also lets me remove all of the dynamic resizing of the bar that I had previously added.

## PR checklist
- [ ] Closes #3391 #3380 
- [ ] Tests added/passed
- [ ] Documentation updated
- [ ] Telemetry [compliance tasks](https://aka.ms/devhome-telemetry) completed for added/updated events

### CODE REVIEW

File: tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs [hunk 1]
```diff
@@ -179,60 +183,79 @@ public void PopulateCommandBar()
...
+        ToolTipService.SetToolTip(button, tool.Name);
 
+        return button;
+    }
+
+    private MenuFlyout CreateMenuFlyout(ExternalTool tool, PinOption pinOption)
+    {
         MenuFlyout menu = new MenuFlyout();
-        menu.Items.Add(tool.IsPinned ? CreateUnPinMenuItem(tool) : CreatePinMenuItem(tool));
+        menu.Items.Add(pinOption == PinOption.Pin ? CreatePinMenuItem(tool) : CreateUnPinMenuItem(tool));
```
Comment (dabhattimsft) [inline tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs:210]: >CreatePinMenuItem(tool) : CreateUnPinMenuItem(tool) [](http://example.com/codeflow?start=52&length=51)

nit: Maybe single CreateMenuItem(tool, bool pin)?

File: tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs [hunk 2]
```diff
@@ -179,60 +183,79 @@ public void PopulateCommandBar()
...
 
-        // If a tool is pinned, we'll add it to the primary commands list, otherwise the secondary commands list
+        return menu;
+    }
+
+    private void AddToolToCommandBar(ExternalTool tool)
+    {
+        // We create 2 copies of the button, one for the primary commands list and one for the secondary commands list.
+        // We're not allowed to put the same button in both lists.
+        AppBarButton pinnedButton = CreateAppBarButton(tool, PinOption.UnPin); // The pinned button should always have the unpin option
```
Comment (dabhattimsft) [inline tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs:220]: >pinnedButton [](http://example.com/codeflow?start=21&length=12)

maybe rename to primaryComamndListButton and secondaryCommandListButton?

**Review by dabhattimsft — APPROVED** [review]

**Review by andreww-msft — APPROVED** [review]

File: tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs (hunk 1 above)
Comment (timkur) [reply tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs:210]: Good suggestion.

File: tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs (hunk 2 above)
Comment (timkur) [reply tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs:220]: Sounds good.

**Pull request merged**